### ISSUE TITLE
$FOO evaluates to '$FOO` rather than value of env var.  (in unmerged PR)

### ISSUE THREAD

**Issue by bobhy** [xonsh/xonsh#3743]
I need team member help completing PR #3742.  I've made a few (really, very few!) changes to xonsh/environ.py and now have broken environment variable reference.  Here's the syndrome:

```
$ $test = 'foo'
$ $test
'foo'    # as expected
$ "abc $test"
"abc $test"    # bogon, expected "abc foo"
$ echo $test
$test  # bogon, expected foo
$ echo ${'te' + 'st'}
foo    # as expected
```

I think I have 1 broken and one working case in both python mode and subproc mode?  

I'm pretty sure this is a parsing issue.  I see this parsing code that looks like it handles the $<name> case, but cannot pierce the veil to see where (and how!) the code generated by `ast.Subscript` actually accesses the environment mapping.  

https://github.com/xonsh/xonsh/blob/582b5a729abcbf50c2ce23305e5d0505df2221db/xonsh/parsers/base.py#L2943-L2949

When it fails, it's apparently leaving the token as a literal, that is not a value it could possibly have gotten by invoking `Env.__getitem__`. However, that is exactly the area I made changes to, so it somehow managed to break things.

You can see the detailed code change #3742.  Test cases for tests_environ.py all work, needed a few tweaks that make sense given the changed behavior.

Any help understanding what's going on here will be good education, even if the PR itself is ultimately rejected.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by laloch** [comment]
You are not right about
```
$ "abc $test"
"abc $test"    # bogon, expected "abc foo"
```
Environment variables within strings are not expanded in Python mode, so the above is correct.

```
$ echo $test
$test  # bogon, expected foo
```
Yeah, you are right about this one and it's quite funny. Your change made the code below to effectively disable the variable expansion.
https://github.com/xonsh/xonsh/blob/5aed9ee5cab85039a5863846b4dd9d2059d824a5/xonsh/tools.py#L98-L104

**Comment by bobhy** [comment]
With this clue, I was able to find the bug.  For the record, the effect of my (botched) PR was, in line 103 above, to return `False` rather than the registered default `True` (defined in xonsh/environ.py) whenever EXPAND_ENV_VARS was unset (which it is, by default). And so, the environment variables were indeed not expanded.  I've found lots of other examples in the Xonsh codebase where hard-coded defaults on `env.get()` no longer match the registered default, will need a different approach to allow the registered default to apply without having an undefined environment variable appear defined when the variable should be set by external code rather than Xonsh.
